### PR TITLE
Wip/catsuryuu/modifications highlighting

### PR DESCRIPTION
**Pull request by catsuryuu**
Bytes marked as modified are exactly the same as these that will be send to server.

### CODE REVIEW

File: src/util/edit.cc [hunk 1]
```diff
@@ -176,8 +172,46 @@ data::BinData EditEngine::bytesValues(size_t pos, size_t size) const {
...
+          result[i] = true;
+        }
+      }
+      ++it;
+      ++next_it;
+    }
+    if (it->fragment_ != nullptr) {
+      for (size_t i = it.key() - pos; i < size; ++i) {
+        result[i] = true;
+      }
```
Comment (mkow) [inline src/util/edit.cc]: Please deduplicate code here. For example, you can fold the for-loops into a lambda taking (start, end) as arguments.

Comment (catsuryuu) [reply src/util/edit.cc]: DONE

Comment (catsuryuu) [reply src/util/edit.cc]: Fixed

File: include/util/edit.h [hunk 2]
```diff
@@ -64,6 +63,9 @@ class EditEngine {
   };
 
   ui::FileBlobModel* original_data_;
+  // TODO(catsuryuu): change to std::map after switching to C++17 (and use
+  // `insert_or_assign`)
+  // Reason: QMap works on `int` type
```
Comment (mkow) [inline include/util/edit.h]: 'QMap works on `int` type' -> 'QMap uses `int` for container size'
Current version may be unclear for a reader, especially that in the next line you use `size_t`.

File: src/util/edit.cc [hunk 3]
```diff
@@ -175,8 +171,38 @@ data::BinData EditEngine::bytesValues(size_t pos, size_t size) const {
...
+std::vector<bool> EditEngine::modifiedPositions(size_t pos, size_t size) const {
+  std::vector<bool> result(size);
+
+  size_t end_pos = pos + size;
+  auto next_it = address_mapping_.upperBound(pos);
+  assert(next_it != address_mapping_.cbegin());
+  auto it = next_it;
+  --it;
+
+  auto setResultTrue = [&it, &result](size_t start, size_t end) {
```
Comment (mkow) [inline src/util/edit.cc]: it's a local variable, so it should be named `set_result_true`

File: include/util/edit.h (hunk 2 above)
Comment (catsuryuu) [reply include/util/edit.h]: Fixed

**Pull request merged**